File: 083_remove_duplicates_from_sorted_list/Solution.java
class ListNode {
    int val;
    ListNode next;
    ListNode(int val) {
        this.val = val;
    }
}

class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) return head;

        ListNode tmp = head;
        while(tmp != null && tmp.next != null) {
            if (tmp.next.val == tmp.val) tmp.next = tmp.next.next;
            else tmp = tmp.next;
        }

        return head;
    }

    public static void main(String[] args) {
        ListNode head = new ListNode(1);
        head.next = new ListNode(1);
        head.next.next = new ListNode(2);
        head.next.next.next = new ListNode(3);
        head.next.next.next.next = new ListNode(3);

        new Solution().deleteDuplicates(head);

        while (head != null) {
            System.out.println(head.val);
            head = head.next;
        }
    }
}

File: 083_remove_duplicates_from_sorted_list/Solution.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if not head: return head

        tmp = head
        while tmp and tmp.next:
            if tmp.next.val == tmp.val: tmp.next = tmp.next.next
            else: tmp = tmp.next

        return head


if __name__ == '__main__':
    head = ListNode(1)
    head.next = ListNode(1)
    head.next.next = ListNode(2)
    head.next.next.next = ListNode(3)
    head.next.next.next.next = ListNode(3)
    Solution().deleteDuplicates(head)

    while head:
        print(head.val)
        head = head.next

